agent/recorder-agent.c

```diff
@@ -0,0 +1,419 @@
+  g_autoptr (GstCaps) stream_caps = NULL;
+  g_autofree gchar *stream_caps_str = NULL;
+  g_autofree gchar *uri = NULL;
+  const gchar *container_type;
+
+  discoverer = gst_discoverer_new (5 * GST_SECOND, &error);
+  g_assert_no_error (error);
+
+  uri = g_strdup_printf ("file://%s", file_path);
+
+  info = gst_discoverer_discover_uri (discoverer, uri, &error);
+  g_assert_no_error (error);
+  g_assert_cmpint (gst_discoverer_info_get_result (info), ==,
+      GST_DISCOVERER_OK);
+
+  stream_info = gst_discoverer_info_get_stream_info (info);
+  stream_caps = gst_discoverer_stream_info_get_caps (stream_info);
+
+  stream_caps_str = gst_caps_to_string (stream_caps);
+  g_debug ("Container file has caps: %s", stream_caps_str);
+
+  g_assert_cmpint (gst_caps_get_size (stream_caps), ==, 1);
+
+  if (g_str_has_suffix (file_path, ".mp4")) {
+    container_type = "video/quicktime";
+  } else if (g_str_has_suffix (file_path, ".ts")) {
+    container_type = "video/mpegts";
+  } else {
+    g_assert_not_reached ();
+  }
+
+  g_assert_cmpstr
+      (gst_structure_get_name (gst_caps_get_structure (stream_caps, 0)), ==,
+      container_type);
+
+  return gst_discoverer_info_get_duration (info);
+}
+
+static void
+stream_connected_cb (HwangsaeRecorder * recorder, HwangsaeRecorderAgent * self)
+{
+  g_warning ("Stream connected");
+}
+
+static void
+file_created_cb (HwangsaeRecorder * recorder, const gchar * file_path,
+    HwangsaeRecorderAgent * self)
+{
+  g_debug ("File %s created", file_path);
+
+  g_assert_false (self->got_file_created_signal);
+  self->got_file_created_signal = TRUE;
+}
+
+static void
+file_completed_cb (HwangsaeRecorder * recorder, const gchar * file_path,
+    HwangsaeRecorderAgent * self)
+{
+  GstClockTime duration = get_file_duration (file_path);
+
+  g_debug ("Finished recording %s, duration %" GST_TIME_FORMAT, file_path,
+      GST_TIME_ARGS (duration));
+
+/*  g_assert_cmpint (labs (GST_CLOCK_DIFF (duration, 5 * GST_SECOND)), <=,
+      GST_SECOND);
+*/
+  g_assert_false (self->got_file_completed_signal);
+  self->got_file_completed_signal = TRUE;
+}
+
+static void
+stream_disconnected_cb (HwangsaeRecorder * recorder,
+    HwangsaeRecorderAgent * self)
+{
+  g_debug ("Stream disconnected");
+}
+
+static void
+hwangsae_recorder_agent_start_recording (HwangsaeRecorderAgent * self,
+    gchar * edge_id)
+{
+  g_autoptr (GError) error = NULL;
+  g_autofree gchar *host =
+      g_settings_get_string (self->settings, "relay-address");
+  guint port = g_settings_get_uint (self->settings, "relay-stream-port");
+  g_autofree gchar *url = NULL;
+
+  if (self->is_recording) {
+    g_warning ("recording already started");
+    return;
+  }
+
+  self->is_recording = TRUE;
+  self->got_file_created_signal = FALSE;
+  self->got_file_completed_signal = FALSE;
+
+  hwangsae_recorder_agent_send_rest_api (self, RELAY_METHOD_START_STREAMING,
+      edge_id);
+
+  url = g_strdup_printf ("srt://%s:%d", host, port);
```

Comment:
You should add `edge_id` into `url` as a part of its query string. Refer [here](https://github.com/xhaakon/hwangsae/blob/c2f13f578632e6cd1c3a300ada02400ec9b519bd/tests/test-relay.c#L139) for how to do it (`username` is your `edge_id`).

Reply:
Thanks.